Rakefile

```
@@ -23,29 +23,52 @@ end
 
 namespace :documentation do
   desc "Update the PLATFORMS.md file with a list of all platforms"
-  task :update_platforms do
+  task update_platforms: :update_json_list do
     File.delete("PLATFORMS.md") if File.exist?("PLATFORMS.md")
     f = File.new("PLATFORMS.md", "w")
     f.write "## Fauxhai Platforms\n\nThis file lists each platform known to Fauxhai and the available versions for each of those platforms. See the ChefSpec documentation for mocking out platforms and platform versions within ChefSpec.\n"
-    Dir.glob("./lib/fauxhai/platforms/**").sort.each do |platform_path|
-      next if platform_path.split("/")[-1] == "chefspec"
-
-      versions = {}
-      Dir.glob(File.join(platform_path, "**.json")).sort.each do |version_path|
-        # skip anything marked as deprecated
-        data = JSON.parse(File.read(version_path))
-        versions[version_path.split("/")[-1].chomp(".json")] = data["deprecated"] ? :deprecated : :supported
-      end
 
-      f.write "\n### #{platform_path.split("/")[-1]}\n\n"
-      versions = versions.sort_by { |v, _s| Gem::Version.new(v) } unless platform_path.split("/")[-1] == "windows" # make sure we're sorted by version and not by strings
-      versions.each do |ver, status|
-        f.write "- #{ver}#{" (deprecated)" if status == :deprecated}\n"
-      end
+    JSON.parse(File.read("platforms.json")).each do |plat, versions|
+      f.write "\n### #{plat}\n\n"
+      versions.each { |ver, data| f.write "- #{ver}#{" (deprecated)" if data["deprecated"]}\n" }
     end
     f.close
   end
 end
 
+desc "Update the platforms.json data file used when a platform can't be found locally"
+task :update_json_list do
+  # get a sorted list of platforms from files in the platforms dir
+  # but skip any hidden files and skip the fake chefspec platform
+  plats = Dir.children("./lib/fauxhai/platforms/").sort
+    .select { |v| !v.match?(/(^\.|chefspec)/) }
+    .to_h { |v| [v, {}] }
+
+  # add version, deprecated status, and file path to each platform
+  plats.each_pair do |plat, _data|
+    ver_data = Dir.glob(File.join("lib/fauxhai/platforms/", plat, "**.json"))
+
+    # we can't properly sort windows versions
+    # but this is a quick and dirty sort that's better than nothing
+    ver_data = if plat == "windows"
+                 ver_data.sort
+               else
+                 ver_data.sort_by { |x| Gem::Version.new(x.split("/")[-1].chomp(".json")) }
+               end
+
+    ver_data.each do |version_path|
+      data = JSON.parse(File.read(version_path))
+      ver = version_path.split("/")[-1].chomp(".json")
+      plats[plat][ver] = {}
+      plats[plat][ver]["deprecated"] = data["deprecated"] ? true : false
```

```suggestion
      plats[plat][ver]["deprecated"] = !!data["deprecated"]
```

The `!!` is probably not strictly necessary but just in case...